lib/application/widgets/theme/rich_text_field.dart

```diff
@@ -0,0 +1,343 @@
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_quill/flutter_quill.dart' as quill;
+import 'package:layoutr/common_layout.dart';
+import 'package:memo/application/constants/dimensions.dart' as dimens;
+import 'package:memo/application/constants/images.dart' as images;
+import 'package:memo/application/constants/strings.dart' as strings;
+import 'package:memo/application/theme/theme_controller.dart';
+import 'package:memo/application/utils/bottom_sheet.dart';
+import 'package:memo/application/widgets/material/asset_icon_button.dart';
+import 'package:memo/application/widgets/theme/custom_button.dart';
+import 'package:memo/application/widgets/theme/themed_container.dart';
+import 'package:tuple/tuple.dart';
+
+/// A [TextField] variation that supports rich text content.
+///
+/// The field supports rich text content by opening a modal when tapped with an embedded `Quill Editor`.
+///
+/// The field also supports presenting rich text content, different from [TextField] which only supports simple
+/// text content. The field content height is limited by [dimens.richTextFieldConstraints].
+class RichTextField extends HookWidget {
+  const RichTextField({required this.modalTitle, required this.placeholder, this.controller, this.focus});
+
+  /// The modal title positioned in the top-left corner of the modal editor.
+  final Widget modalTitle;
+
+  /// The placeholder used in the rich text editor when it has no content.
+  final String placeholder;
+
+  /// Controls the text content being edited.
+  final TextEditingController? controller;
```

Comment:
What do you think about using the known `TextEditingController` instead a custom one? I realized we have the same use-cases that a normal textfield has. The difference is that our `text` property will be the rich text JSON instead a normal text.